## Data prep

In [54]:
# Load libraries

import pandas as pd
import numpy as np
import glob
pd.options.plotting.backend = "plotly"
import plotly.io; plotly.io.renderers.default = "notebook_connected"

In [2]:
# Merge files and delete duplicates

files = glob.glob('raw data/*.csv')
all_dfs = []
for f in files:
    _df = pd.read_csv(f, parse_dates = ['DATE'],
                      dtype = {'PER/ACRE':str, 'ASS VAL':str},
                      na_values=['', ' ', 'N/A'])
    all_dfs.append(_df)
    
df = pd.concat(all_dfs, ignore_index=True)
df['ASS VAL'] = df['ASS VAL'].str.replace(',','').astype(float)
df['PER/ACRE'] = df['PER/ACRE'].str.replace(',','').astype(float)

df = df.drop_duplicates()

In [3]:
# Standardize LSDs 

df['LEGAL_STD'] = df.LEGAL.str.replace('LSD 0', 'LSD')

In [4]:
# Calculate price difference between sales

df = df.sort_values(['LEGAL_STD', 'DATE'])
df['PRICE_CHANGE'] = df.groupby('LEGAL_STD').PRICE.diff()
df['ACRE_PRICE_CHANGE'] = df.groupby('LEGAL_STD')['PER/ACRE'].diff()

## Analysis

In [12]:
# Define display columns

cols = ['LEGAL', 'RM', 'DATE', 'PRICE', 'PER/ACRE', 'VENDOR', 'PURCHASER']

#### Basic stats

Number of records

In [17]:
len(df)

38451

Time span of data

In [21]:
df.DATE.describe()

<ipython-input-21-4ac38c483eeb>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  df.DATE.describe()


count                   38451
unique                   5873
top       2014-04-30 00:00:00
freq                       98
first     1993-10-02 00:00:00
last      2021-02-05 00:00:00
Name: DATE, dtype: object

15 biggest single sales

In [13]:
df.sort_values('PRICE', ascending = False).head(15)[cols]

LEGAL                          RM       DATE    PRICE  \
20004     SW-07-28-09-W2              275 - Insinger 2019-12-10  4500000   
21543     NE-27-28-08-W2              275 - Insinger 2005-09-09  4051350   
27818     NW-01-32-08-W2              305 - Invermay 2002-10-04  4000000   
2031      NE-03-08-23-W2              070 - Key West 2002-10-04  4000000   
17185  Parcel# 164702842  250 - Last Mountain Valley 2019-12-09  3762000   
27667     SE-12-33-07-W2              305 - Invermay 2004-05-13  3700000   
1921      SE-18-08-22-W2              070 - Key West 2003-06-18  3550000   
1922      NW-11-08-22-W2              070 - Key West 2003-06-18  3550000   
17184     NW-24-26-22-W2  250 - Last Mountain Valley 2019-12-09  3056400   
32049     NW-08-52-16-W2           488 - Torch River 2020-11-30  2526100   
26418     SW-24-33-07-W2              305 - Invermay 2019-12-10  2250000   
26422     SW-25-33-07-W2              305 - Invermay 2019-12-10  1800000   
20006     SE-28-28-09-W2              275 - Insinger 2019-12-10  1800000   
14250     NW-21-25-13-W2              247 - Kellross 2019-12-09  1800000   
26420     NE-14-33-07-W2              305 - Invermay 2019-12-10  1800000   

        PER/ACRE                          VENDOR  \
20004    3132.83                    John Zuchkan   
21543   25320.94  Yellowhead Pork Producers Ltd.   
27818   25000.00              Big Sky Farms Inc.   
2031    25157.23              Big Sky Farms Inc.   
17185  836000.00          Victoria & Kent Rhodes   
27667   23125.00              Big Sky Farms Inc.   
1921    22327.04              Big Sky Farms Inc.   
1922    22327.04              Big Sky Farms Inc.   
17184    2147.06                  Terry Hamilton   
32049   17360.32  TOPSOIL FARM LAND MANAGEMENT (   
26418    1575.10                    Daniel Sliva   
26422    1259.29                    Daniel Sliva   
20006    1284.93            John & Nancy Zuchkan   
14250    1262.63                      Julia Wahl   
26420    1258.42                    Daniel Sliva   

                            PURCHASER  
20004              John Zuchkan et al  
21543              BIG SKY FARMS INC.  
27818     PRAIRIE HOG PRODUCTION INC.  
2031      PRAIRIE HOG PRODUCTION INC.  
17185               Kent Rhodes et al  
27667     PRAIRIE HOG PRODUCTION INC.  
1921   BIG SKY FINISHERS (NO. 1) INC.  
1922   BIG SKY FINISHERS (NO. 1) INC.  
17184           Murray Hamilton et al  
32049       J & J WINNY HOLDINGS LTD.  
26418              Daniel Sliva et al  
26422              Daniel Sliva et al  
20006              John Zuchkan et al  
14250                 Marv Wahl et al  
26420              Daniel Sliva et al

15 most expensive per-acre sales


In [14]:
df.sort_values('PER/ACRE', ascending = False).head(15)[cols]

LEGAL                 RM       DATE   PRICE    PER/ACRE  \
14548  Parcel# 152172299     247 - Kellross 2018-01-02   50000  5000000.00   
14555  Parcel# 152172299     247 - Kellross 2017-12-01   42450  4245000.00   
33023  Parcel# 135354898  488 - Torch River 2017-10-03  370000  3363636.36   
32909  Parcel# 203329201  488 - Torch River 2018-02-13  479400  2996250.00   
33019  Parcel# 145200945  488 - Torch River 2017-10-19  380000  2923076.92   
32937  Parcel# 135355079  488 - Torch River 2018-01-15  300000  2727272.73   
32741  Parcel# 135354843  488 - Torch River 2018-07-25  290000  2636363.64   
32191  Parcel# 153789182  488 - Torch River 2020-05-11  250000  2272727.27   
32271  Parcel# 203329199  488 - Torch River 2020-02-18  215000  1791666.67   
32114  Parcel# 135355068  488 - Torch River 2020-10-20  170000  1545454.55   
22476  Parcel# 163503008    276 - Foam Lake 2020-09-28  595000  1525641.03   
33206  Parcel# 145200978  488 - Torch River 2017-01-19  195000  1500000.00   
32178  Parcel# 135354944  488 - Torch River 2020-07-07  144500  1313636.36   
32176  Parcel# 135354977  488 - Torch River 2020-07-07  144500  1313636.36   
28732  Parcel# 113435399       336 - Sasman 2018-11-06  175000  1250000.00   

                               VENDOR                       PURCHASER  
14548              VILLAGE OF LESTOCK  RURAL MUNICIPALITY OF KELLROSS  
14555        CORNERSTONE CREDIT UNION              VILLAGE OF LESTOCK  
33023         Quintin & Leanne Wismer   Quint Stacey & Kari Stevenson  
32909            Randy & Shirley Zary            Randy & Shirley Zary  
33019             Donald & Renee Mohr           Katie & Jesse Hupaelo  
32937                     Cindy Tyson            TREVOR & CINDY TYSON  
32741          Brad & Jill McClements                 Brad McClements  
32191              Aaron & Kara Krahn                TAMMY STANKOWSKI  
32271            Randy & Shirley Zary    Christopher & Nancy Komperdo  
32114                      Debra Gane         Debra & Scott Remeshylo  
22476                  MICHEAL BRAHAM         Leonard & Laurie Kayter  
33206               Garry Nachtegaele             VERA & LUCAS HEINZE  
32178  Marion McComb & Shirley Newsom            TREVOR & BRENDA MINO  
32176  Marion McComb & Shirley Newsom            TREVOR & BRENDA MINO  
28732        William & Shelley Rowein                  William Rowein

15 buyers who spent the most and the amounts they paid.


In [83]:
df.groupby('PURCHASER').sum().PRICE.sort_values(ascending = False).head(15)

PURCHASER
ANDJELIC LAND INC.                27268145
J & J WINNY HOLDINGS LTD.         23954578
Topsoil Farm Land Mgmt (II)       22688525
Maxcrop Landing Ltd.              16124110
Murray Hamilton et al             15772000
INPUT CAPITAL CORP.               12505374
PRAIRIE HOG PRODUCTION INC.       11700000
Robert Andjelic                   10241000
HRM the Queen (Canada)             9577731
CAROLYNN WALLIN                    9514690
John Zuchkan et al                 9000000
102053876 SASKATCHEWAN LTD.        8805756
SS UNIMIX AGRICULTURE CORPORAT     7696200
211413 Saskatchewan Inc.           7412232
Marv Wahl et al                    7200000
Name: PRICE, dtype: int64

15 buyers who bought the most acres and how many acres they got. It looks like the government is identified in differnet ways (HMQ, HQM(Canada), etc. These would need to be standardized to get more accurate totals.

In [84]:
df.groupby('PURCHASER').sum().ACRES.sort_values(ascending = False).head(15)

PURCHASER
HRM the Queen (Canada)         46662.89
Maxcrop Landing Ltd.           29607.50
ANDJELIC LAND INC.             26875.89
211413 Saskatchewan Inc.       20628.33
Topsoil Farm Land Mgmt (II)    19880.04
STAN DELORME, ET AL            17310.85
HRM the Queen (Sask)           16212.33
Robert Andjelic                14974.94
Murray Hamilton et al          14288.60
J & J WINNY HOLDINGS LTD.      13604.51
HMQ                            11664.65
HMQ (Canada)                   10429.68
HMQ(Canada)                    10411.45
DOUGLAS & BRENDA HANNAH        10032.35
COWESSESS TLE HOLDINGS INC.     9821.00
Name: ACRES, dtype: float64

Biggest change in per-acre price from one sale to the next. It seems like a lot of these lots stayed in the family or moved between farms.

In [27]:
df.sort_values('PRICE_CHANGE', ascending = False).head(15)[cols + ['PRICE_CHANGE']]

LEGAL                          RM       DATE    PRICE  \
20004     SW-07-28-09-W2              275 - Insinger 2019-12-10  4500000   
21543     NE-27-28-08-W2              275 - Insinger 2005-09-09  4051350   
27818     NW-01-32-08-W2              305 - Invermay 2002-10-04  4000000   
2031      NE-03-08-23-W2              070 - Key West 2002-10-04  4000000   
27667     SE-12-33-07-W2              305 - Invermay 2004-05-13  3700000   
1922      NW-11-08-22-W2              070 - Key West 2003-06-18  3550000   
1921      SE-18-08-22-W2              070 - Key West 2003-06-18  3550000   
17185  Parcel# 164702842  250 - Last Mountain Valley 2019-12-09  3762000   
17184     NW-24-26-22-W2  250 - Last Mountain Valley 2019-12-09  3056400   
33703     NW-08-52-16-W2           488 - Torch River 2015-06-03  1789155   
20006     SE-28-28-09-W2              275 - Insinger 2019-12-10  1800000   
5870      NE-32-20-05-W2               184 - Grayson 2019-12-09  1620000   
5878      SE-32-20-05-W2               184 - Grayson 2019-12-09  1620000   
32358     SW-20-51-14-W2           488 - Torch River 2019-12-09  1566000   
17173     SE-25-26-22-W2  250 - Last Mountain Valley 2019-12-09  1436400   

        PER/ACRE                          VENDOR  \
20004    3132.83                    John Zuchkan   
21543   25320.94  Yellowhead Pork Producers Ltd.   
27818   25000.00              Big Sky Farms Inc.   
2031    25157.23              Big Sky Farms Inc.   
27667   23125.00              Big Sky Farms Inc.   
1922    22327.04              Big Sky Farms Inc.   
1921    22327.04              Big Sky Farms Inc.   
17185  836000.00          Victoria & Kent Rhodes   
17184    2147.06                  Terry Hamilton   
33703   12295.75            Torch River Agr Corp   
20006    1284.93            John & Nancy Zuchkan   
5870     1133.07           Herbert & Monica Haas   
5878     1124.72           Herbert & Monica Haas   
32358    1199.17                 Emily Backstrom   
17173    1006.11                  Terry Hamilton   

                            PURCHASER  PRICE_CHANGE  
20004              John Zuchkan et al     4000000.0  
21543              BIG SKY FARMS INC.     3979350.0  
27818     PRAIRIE HOG PRODUCTION INC.     3958000.0  
2031      PRAIRIE HOG PRODUCTION INC.     3940000.0  
27667     PRAIRIE HOG PRODUCTION INC.     3628000.0  
1922   BIG SKY FINISHERS (NO. 1) INC.     3496000.0  
1921   BIG SKY FINISHERS (NO. 1) INC.     3488000.0  
17185               Kent Rhodes et al     3344000.0  
17184           Murray Hamilton et al     2716800.0  
33703     Topsoil Farm Land Mgmt (II)     1714787.0  
20006              John Zuchkan et al     1600000.0  
5870             JBH FARMS LTD. et al     1595000.0  
5878             JBH FARMS LTD. et al     1440000.0  
32358  CONCENTRA TRUST / LA SOCIETE D     1392000.0  
17173           Murray Hamilton et al     1386400.0

Total sales per year


In [55]:
df.groupby(df.DATE.dt.year).PRICE.sum().plot.bar()

Average change in per-acre price per year

In [56]:
df.groupby(df.DATE.dt.year).ACRE_PRICE_CHANGE.mean().plot.bar()

2014 is a huge outlier, let's take a close look. Here are the 15 sales that year with the biggest price change

In [28]:
big_changes_2014 = df[df.DATE.dt.year == 2014].sort_values('ACRE_PRICE_CHANGE', ascending = False).head(15)
big_changes_2014[cols]

LEGAL                          RM       DATE   PRICE  \
12159      SW-13-24-07-W2               215 - Stanley 2014-09-25   72000   
29172   LSD 9-09-34-12-W2                336 - Sasman 2014-12-08   12520   
12161      NW-13-24-07-W2               215 - Stanley 2014-09-25   60100   
34094  LSD 11-13-52-19-W2           488 - Torch River 2014-04-01  600000   
33947      SW-16-51-14-W2           488 - Torch River 2014-11-13  535500   
6314       NW-20-21-05-W2               184 - Grayson 2014-03-13  932500   
9488       SW-03-22-03-W2             213 - Saltcoats 2014-03-05  896400   
17723      SW-15-25-22-W2  250 - Last Mountain Valley 2014-08-20  520000   
6302       NE-19-21-05-W2               184 - Grayson 2014-03-13  437500   
23303      NE-04-30-11-W2             276 - Foam Lake 2014-03-05  416500   
26794      SW-15-33-08-W2              305 - Invermay 2014-04-10  500000   
26766      SW-01-32-07-W2              305 - Invermay 2014-05-28  260556   
3662       NW-12-11-22-W2            100 - Elmsthorpe 2014-04-15  311000   
552        SW-27-07-22-W2              070 - Key West 2014-11-03  200000   
33984      SE-07-53-17-W2           488 - Torch River 2014-09-29  295000   

        PER/ACRE                        VENDOR                       PURCHASER  
12159  720000.00             Donald Tebb et al           Donald & Noeleen Tebb  
29172  178857.14          William & Mary Kulyk                   William Kulyk  
12161   15774.28             Donald Tebb et al           Noeleen & Donald Tebb  
34094    8854.78                   Carman Long               Carman Long et al  
33947    9916.67       Curtis & Brenda Deutsch               JODY & SHONA FUNK  
6314     5835.42                 Randy Thiedig  SS UNIMIX AGRICULTURE CORPORAT  
9488     5603.55         STEPHEN & LINDA ROPER  SS UNIMIX AGRICULTURE CORPORAT  
17723    3242.91      Barrie & Barbara Gwillim                 Barbara Gwillim  
6302     2762.52                 Randy Thiedig  SS UNIMIX AGRICULTURE CORPORAT  
23303    2608.18     Ronald & Rita Zadorozniak          Julian Evanochko et al  
26794    3161.96  Barry & Justin & Korwin Enge            Justin & Korwin Enge  
26766    3290.26                Russell Shukin          GENEVIEVE SHUKIN et al  
3662     1943.02           Frobisher Farm Ltd.                         HUAI LI  
552      3352.89             Bryan Smith et al               BRYAN SMITH et al  
33984    1870.05      Wayne & Florence McAuley             Dennis & Wanda Wood

Let's isolate those properties and look at past sales. Look at property LSD 9-09-34-12-W2. It was sold by William Kulyk for \\$1 in 2012, then bought by William & Mary Kulyk for \\$1, then bought back by William Kulyk for \\$12520 in 2014. Is this anomalous?

In [98]:
df[df.LEGAL_STD.isin(big_changes_2014.LEGAL_STD)][['LEGAL_STD', 'DATE', 'PRICE', 'ACRE_PRICE_CHANGE', 'VENDOR', 'PURCHASER']]

LEGAL_STD       DATE   PRICE  ACRE_PRICE_CHANGE  \
36881  LSD 11-13-52-19-W2 2000-09-15   16500                NaN   
34094  LSD 11-13-52-19-W2 2014-04-01  600000            8442.28   
29559   LSD 9-09-34-12-W2 2012-04-19       1                NaN   
29367   LSD 9-09-34-12-W2 2013-05-22       1               0.00   
29172   LSD 9-09-34-12-W2 2014-12-08   12520          178842.85   
24926      NE-04-30-11-W2 2003-12-08   40000                NaN   
23303      NE-04-30-11-W2 2014-03-05  416500            2358.18   
8703       NE-19-21-05-W2 1994-12-20   26000                NaN   
6302       NE-19-21-05-W2 2014-03-13  437500            2600.02   
4849       NW-12-11-22-W2 2002-02-12   10000                NaN   
3662       NW-12-11-22-W2 2014-04-15  311000            1880.52   
3669       NW-12-11-22-W2 2014-04-15  311000               0.00   
13228      NW-13-24-07-W2 2003-06-09   46000                NaN   
12158      NW-13-24-07-W2 2014-09-25   53700              58.58   
12161      NW-13-24-07-W2 2014-09-25   60100           15424.56   
8700       NW-20-21-05-W2 1994-12-20   32000                NaN   
6314       NW-20-21-05-W2 2014-03-13  932500            5635.42   
36945      SE-07-53-17-W2 2000-06-23   20000                NaN   
33984      SE-07-53-17-W2 2014-09-29  295000            1745.05   
27006      SW-01-32-07-W2 2012-06-19   45000                NaN   
27003      SW-01-32-07-W2 2012-06-29   90000             568.26   
26766      SW-01-32-07-W2 2014-05-28  260556            2153.75   
26767      SW-01-32-07-W2 2014-05-28  521112               0.21   
26771      SW-01-32-07-W2 2014-05-28  521112               0.00   
10432      SW-03-22-03-W2 2005-07-22  249500                NaN   
9488       SW-03-22-03-W2 2014-03-05  896400            4044.17   
13229      SW-13-24-07-W2 2003-06-09   45000                NaN   
12156      SW-13-24-07-W2 2014-09-25   47500              15.22   
12159      SW-13-24-07-W2 2014-09-25   72000          719701.76   
19806      SW-15-25-22-W2 1995-09-01   45270                NaN   
17723      SW-15-25-22-W2 2014-08-20  520000            2959.97   
17413      SW-15-25-22-W2 2017-06-16  359220           -1002.69   
27744      SW-15-33-08-W2 2003-03-17  300000                NaN   
27590      SW-15-33-08-W2 2005-04-20  150000            -949.36   
26794      SW-15-33-08-W2 2014-04-10  500000            2212.59   
37461      SW-16-51-14-W2 1998-09-01  160000                NaN   
33947      SW-16-51-14-W2 2014-11-13  535500            8099.11   
1506       SW-27-07-22-W2 2008-03-12       0                NaN   
1503       SW-27-07-22-W2 2008-03-17  144500                NaN   
552        SW-27-07-22-W2 2014-11-03  200000            1863.20   
391        SW-27-07-22-W2 2016-10-26  190500            -159.26   

                             VENDOR                       PURCHASER  
36881        Frank/Edna/David Pagan                     Carman Long  
34094                   Carman Long               Carman Long et al  
29559                 William Kulyk                 LAWRENCE HASKEY  
29367               LAWRENCE HASKEY            William & Mary Kulyk  
29172          William & Mary Kulyk                   William Kulyk  
24926              Olga Zadorozniak       Ronald & Rita Zadorozniak  
23303     Ronald & Rita Zadorozniak          Julian Evanochko et al  
8703                Ottilia Thiedig                   Randy Thiedig  
6302                  Randy Thiedig  SS UNIMIX AGRICULTURE CORPORAT  
4849                  Dwite Demerse           Dwite & Linda Demerse  
3662            Frobisher Farm Ltd.                         HUAI LI  
3669          Dwite & Linda Demerse             Frobisher Farm Ltd.  
13228                 Steve Somogyi               Donald Tebb et al  
12158             Donald Tebb et al           Donald & Noeleen Tebb  
12161             Donald Tebb et al           Noeleen & Donald Tebb  
8700                Ottilia Thiedig                   Randy Thiedig  
6314       

Let's look at stats by RM: number of sales, total amount of sales, and average price per sale. Saltcoats seems to be an outlier with a higher average sales price.

In [33]:
import seaborn as sns

cm = sns.light_palette("blue", as_cmap=True)

stats_by_rm = df.groupby('RM').PRICE.agg(['count','sum', 'mean']).sort_values('mean', ascending = False)
stats_by_rm.style.background_gradient(cmap=cm)

Now let's explore the relationship between acreage and price over time. The chart below show each sale in the data, coloure by RM. Use the slider to cycle through the years. Hover over dots for details.

For the most part, there's a straight relationship between acre and price. Most parcels are around 140-170 acres. But in some years, in some RMs, we see big outliers.

Examples:

* In 2008, a lot of parcels were sold for \\$1.
* In 2011 in Invernay, there were 160-acre parcels sold far above the average for that size.
* Torch River seems to have more under-valued parcels.

In [57]:
import plotly.express as px

px.scatter(df, x = 'ACRES', y = 'PRICE', color = 'RM', 
           animation_frame = sorted(df.DATE.dt.year), range_y = (1, df.PRICE.max()+10000),
           range_x = (1, df.ACRES.max()+100), log_x = True, log_y = True, height = 600,
           template = 'plotly_white')

In [58]:
%%html
<script src="https://cdn.rawgit.com/parente/4c3e6936d0d7a46fd071/raw/65b816fb9bdd3c28b4ddf3af602bfd6015486383/code_toggle.js"></script>